src/Stratis.Bitcoin.Tests/Connection/ConnectionManagerTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Configuration.Settings;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Tests.Common;
+using Stratis.Bitcoin.Tests.Common.Logging;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.Connection
+{
+    public class ConnectionManagerSettingsTest : LogsTestBase
+    {
+        private readonly Mock<IConnectionManager> connectionManager;
+        private ConnectionManagerController controller;
+        private readonly Mock<ILoggerFactory> mockLoggerFactory;
+
+        public ConnectionManagerSettingsTest()
+        {
+            this.connectionManager = new Mock<IConnectionManager>();
+            this.mockLoggerFactory = new Mock<ILoggerFactory>();
+            this.mockLoggerFactory.Setup(i => i.CreateLogger(It.IsAny<string>())).Returns(new Mock<ILogger>().Object);
+            this.connectionManager.Setup(i => i.Network)
+                .Returns(KnownNetworks.StratisTest);
+            this.controller = new ConnectionManagerController(this.connectionManager.Object, this.LoggerFactory.Object);
+        }
+
+        [Fact]
+        public void WhiteListedEndpoint_CanBeMappedAndRoutedTo_OtherNode()
+        {
+            NodeServerEndpoint endpointA = null;
+            NodeServerEndpoint endpointB = null;
+            NodeServerEndpoint endpointOut = null;
+            
+            var connectionManagerSettings = new ConnectionManagerSettings(NodeSettings.Default(this.Network));
+
+            // 127.0.0.1:16178 == 0.0.0.0:16178
+            endpointA = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("127.0.0.1"), 16178), false);
+            endpointB = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), 16178), false);
+            connectionManagerSettings.Listen = new List<NodeServerEndpoint>() { endpointB};
+            connectionManagerSettings.Port = 16178;
+            Assert.True(endpointA.CanBeMappedTo(connectionManagerSettings.Listen, out endpointOut));
```

Comment:
// Correct condition should be 127.0.0.1:16178 != 0.0.0.0:16178
// If we have bound FullNode to all local interfaces (i.e. 0.0.0.0:16178) but want to whilte list peers connecting from localhost (i.e. 127.0.0.1:16178)
Assert.False(...